cli/cli-client/src/main/java/org/infinispan/cli/commands/CliCommand.java

```diff
@@ -30,7 +32,17 @@ public CommandResult execute(ContextAwareCommandInvocation invocation) throws Co
          invocation.println(invocation.getHelpInfo());
          return CommandResult.SUCCESS;
       }
-      return exec(invocation);
+      try {
+         return exec(invocation);
+      } catch (CommandException e) {
+         invocation.getShell().writeln(ANSI.YELLOW_TEXT + e.getLocalizedMessage() + ANSI.DEFAULT_TEXT);
+         return CommandResult.FAILURE;
+      } catch (Throwable e) {
+         // These are unhandled, show them in red
+         Throwable cause = Util.getRootCause(e);
```

Comment:
well, I was expecting a new method in `org.infinispan.cli.Context[Impl]` with [1]  and just invoke `invocation.getContext().new_method(e)`... but I guess this works as well :) 

```java
TerminalString error = new TerminalString(Util.getRootCause(e).getLocalizedMessage(), new TerminalColor(Color.RED, Color.DEFAULT, Color.Intensity.BRIGHT));
shell.writeln(error.toString());
```